spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+
+=== Specify Config Profile
+
+The config profile can be specified via the property `mp.config.profile`, which can be set in any of the configuration sources. The value of the property can contain only characters that are valid for config property names. 
+This is because the name of the profile is directly stored in the name of the config property. It can be set when starting your application. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The value of the property `mp.config.profile` shouldn't be updated after the application is started. It's only read once during the application start up. If the property value is modified afterwards, the behavior is undefined and any changes to its value made later can be ignored by the implementation. 
+
+The value of the property `mp.config.profile` specifies a single active profile. Only one profile can be active at a time. Commas in the value have no special meaning. For example, if the value of `mp.config.profile` is `testing,live`,  a single profile named `testing,live` is active instead of two active profiles. 
+If the property `mp.config.profile` is specified in multiple config sources, the value of the property is determined following the same rules as other configuration properties, which means the value in the config source with the highest ordinal wins.
+
+=== How Config Profile works
```

Comment:
This section isn't as clear as I'd hope. The way I understand it profiles do not have an effect beyond their source. If a source has either the original property or the profile specific property and the highest ordinal it wins (as usual) whereby naturally the profile specific property (if present) does override the original property value within that winning source.

Reply:
@jbee your undestanding is correct. Within the same config source, the active profile specific property overrides the original property value. Overall, the property from the highest ordinal wins no matter whether it is the original property or the active profile property.